library/lvm_gensym.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python 
+
+from ansible.module_utils.basic import AnsibleModule 
+from ansible.module_utils import facts
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: lvm_gensym
+short_description: Generate default names for lvm variables
+version_added: "2.4"
+description: 
+    - "Module accepts two input strings consisting of a file system type and 
+       a mount point path, and outputs names based on system information"
+options:
+    fs_type:
+        description:
+            - String describing the desired file system type 
+        required: true 
+    mount:
+        description:
+            - String describing the mount point path 
+        required: true
+author: 
+    - Tim Flannagan (tflannag@redhat.com)
+'''
+
+EXAMPLES = '''
+- name: Generate names 
+  lvm_gensym:
+    fs_type: "{{ fs_type }}"
+    mount: "{{ mount_point }}"
+  register: lvm_results
+  when: lvm_vg == "" && mount_point != "" and fs_type != ""
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+
+lv_name:
+    description: The default generated name for an unspecified logical volume 
+    type: str
+'''
+
+def is_unique(name, used_names):
+    if name not in used_names:
+        return True 
+    
+    return False 
+
+def init_vg_name(host_name, vg_names):
+    def get_os_name():    
+        for line in open('/etc/os-release').readlines():
+            if not line.find('ID='):
+                name = line[3:]
```

Comment:
You should probably `break` after setting name to save the time/work of looking at the rest of the file.